backend/plugin/db/mysql/role.go

```diff
@@ -388,9 +391,23 @@ func (driver *Driver) getInstanceRoles(ctx context.Context) ([]*storepb.Instance
 	`
 	var instanceRoles []*storepb.InstanceRoleMetadata
 	roleRows, err := driver.db.QueryContext(ctx, query)
-
 	if err != nil {
-		return nil, util.FormatErrorWithQuery(err, query)
+		// RDS may not have the permission to query mysql.user.
+		log.Warn("failed to query mysql.user", zap.Error(err))
+		// try reading from information_schema.user_attributes
+		query := `
+		SELECT
+			user,
+			host
+		FROM information_schema.user_attributes
+		WHERE user NOT LIKE 'mysql.%'
+		`
+		roleRows, err = driver.db.QueryContext(ctx, query)
```

Comment:
Move to a separate function or use a different role1Rows because we need to defer Close() later.